tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -44,10 +45,10 @@ public void ChangeStreamExample1()
             .Start();
 
             // Start Changestream Example 1
-            var enumerator = inventory.Watch().ToEnumerable().GetEnumerator();
-            enumerator.MoveNext();
-            var next = enumerator.Current;
-            enumerator.Dispose();
+            var cursor = inventory.Watch();
+            while (cursor.MoveNext() && cursor.Current.Count() == 0) { } // keep calling MoveNext until we've read the first batch
```

Comment:
These examples are fundamentally flawed.

What application would write code to read ONLY the first change document from a change stream.

What the example *should* be is code showing how to loop while processing multiple change documents.

But... I guess this is a correct implementation of a fundamentally flawed example.

Reply:
Robert, I disagree with the severity of your assessment.  Looking at the docs page at https://docs.mongodb.com/manual/changeStreams/ I can see why it was done this way.  There are multiple examples showing different aspects of change streams, and it was probably seen as too distracting to show a loop for each example.

However, the .NET driver is an anomaly in that it exposes a batch-oriented cursor.  In the other driver examples you can imagine, if you know the language, that you would just call next() in a loop using the iteration idiom of the language.  But in the .NET driver it's less obvious what's going on.  The call to `cursor.Current.First()` looks more awkward than the corresponding call to `cursor.next()` in the Java examples.

Given the current constraints, would it make more sense to remove the while loop as well, and just do:

     cursor.MoveNext();
     cursor.Current.First();
     cursor.Dispose();

?  

I think this is also worth a DOCSP ticket expressing your concerns on the overall structure of the examples.  I'm sure they'd appreciate our help in making it better.

Reply:
That code assumes there is already a change document pending in the change stream.

Otherwise First will throw an exception on an empty batch.

My assertion that the examples are flawed is based on my opinion that no application should *ever* write code that looks like this. By providing this code as an example we are telling our users that this is the way change streams should be processed.

I'll re-review quickly and LGTM since this does in fact correctly implement what we were asked for.